.github/workflows/validate.yml

```diff
@@ -0,0 +1,40 @@
+name: Validate
+# Lints, builds, and tests the project whenever a pull request is opened or merged
+
+on:
+  pull_request:
+    branches: [main]
+  push:
+    branches: [main]
+  workflow_dispatch:
+
+jobs:
+  validate:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Setup node
+        uses: actions/setup-node@v3
+        with:
+          node-version-file: '.nvmrc'
+
+      - name: Install dependencies
+        run: npm ci
+
+      - name: Generate version
+        run: npm run export-version
+
+      - name: Lint
+        run: npm run lint
+
+      - name: Build
+        run: npm run build
+
+      - name: Tests
+        run: npm test
+
+      # - name: Coverage
+      #   uses: ArtiomTr/jest-coverage-report-action@v2.2.2
```

Comment:
@AverageHelper is there a reason this is commented out?

Reply:
For reference until I figure out how to implement it such that it works in external PRs. Can probs remove it, I'm sure we'll get around to engineering that eventually